otx/algorithms/visual_prompting/adapters/pytorch_lightning/datasets/dataset.py

```diff
@@ -0,0 +1,332 @@
+            )
+        )
+        item = self.transform(item)
+        return item
+
+    def convert_polygon_to_mask(self, shape: Polygon, width: int, height: int) -> np.ndarray:
+        """Convert polygon to mask.
+
+        Args:
+            shape (Polygon): Polygon to convert.
+            width (int): Width of image.
+            height (int): Height of image.
+
+        Returns:
+            np.ndarray: Generated mask from given polygon.
+        """
+        polygon = ShapeFactory.shape_as_polygon(shape)
+        contour = [[int(point.x * width), int(point.y * height)] for point in polygon.points]
+        gt_mask = np.zeros(shape=(height, width), dtype=np.uint8)
+        gt_mask = cv2.drawContours(gt_mask, np.asarray([contour]), 0, 1, -1)
+        return gt_mask
+
+    def generate_bbox(self, x1: int, y1: int, x2: int, y2: int, width: int, height: int) -> List[int]:  # noqa: D417
+        """Generate bounding box.
+
+        Args:
+            x1, y1, x2, y2 (int): Bounding box coordinates. # type: ignore
+            width (int): Width of image.
+            height (int): Height of image.
+
+        Returns:
+            List[int]: Generated bounding box.
+        """
+
+        def get_randomness(length: int) -> int:
+            if self.offset_bbox == 0:
+                return 0
+            return np.random.normal(0, min(length * 0.1, self.offset_bbox))
+
+        bbox = [
+            max(0, x1 + get_randomness(width)),
+            max(0, y1 + get_randomness(height)),
+            min(width, x2 + get_randomness(width)),
+            min(height, y2 + get_randomness(height)),
+        ]
+        return bbox
+
+    def generate_bbox_from_mask(self, gt_mask: np.ndarray, width: int, height: int) -> List[int]:
+        """Generate bounding box from given mask.
+
+        Args:
+            gt_mask (np.ndarry): Mask to generate bounding box.
+            width (int): Width of image.
+            height (int): Height of image.
+
+        Returns:
+            List[int]: Generated bounding box from given mask.
+        """
+        y_indices, x_indices = np.where(gt_mask == 1)
+        x_min, x_max = np.min(x_indices), np.max(x_indices)
+        y_min, y_max = np.min(y_indices), np.max(y_indices)
+        return self.generate_bbox(x_min, y_min, x_max, y_max, width, height)
+
+
+class OTXVisualPromptingDataModule(LightningDataModule):
+    """Visual Prompting DataModule.
+
+    Args:
+        config (Union[DictConfig, ListConfig]): Configuration.
+        dataset (DatasetEntity): Dataset entity.
+    """
+
+    def __init__(self, config: Union[DictConfig, ListConfig], dataset: DatasetEntity) -> None:
+        super().__init__()
+        self.config = config
+        self.dataset = dataset
+
+        self.train_otx_dataset: DatasetEntity
+        self.val_otx_dataset: DatasetEntity
+        self.test_otx_dataset: DatasetEntity
+        self.predict_otx_dataset: DatasetEntity
+
+    def setup(self, stage: Optional[str] = None) -> None:
```

Comment:
It's just a question. Does visual prompting use fixed data pipeline?

Reply:
As I addressed, it uses fixed data pipeline, but it can be moved in sam module for export.
If not, I'll combine them.